### ISSUE TITLE
CT.LAMBDA.PR.2 triggered unexpectedly

### ISSUE THREAD

**Issue by cmaster11** [cdklabs/cdk-validator-cfnguard#390]
Hi, I've encountered the issue where the CT.LAMBDA.PR.2 is triggered by the plugin but not when deploying code with the plugin disabled (the rule is enabled in our AWS account Control Tower).

The following CF template code triggers the rule for some reason that is unknown to me. To be clear, this code is introduced by the CDK natively at https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-apigatewayv2-integrations/lib/http/lambda.ts#L55 whenever you use the Lambda integration for an API gateway.

```
const integration = new LambdaIntegration(myLambdaFunction);
const method = route.addMethod(httpMethod, integration); // <-- Call that adds the permission
```

```json
{ 
  "MyPermissions": {
   "Type": "AWS::Lambda::Permission",
   "Properties": {
    "Action": "lambda:InvokeFunction",
    "FunctionName": {
     "Fn::GetAtt": [
      "SomeAttributeXXXX",
      "Arn"
     ]
    },
    "Principal": "apigateway.amazonaws.com",
    "SourceArn": {
     "Fn::Join": [
      "",
      [
       "arn:aws:execute-api:us-east-2:123456789012:",
       {
        "Ref": "Gatewayblablabal"
       },
       "/",
       {
        "Ref": "Gateway123456"
       },
       "/POST/report"
      ]
     ]
    }
   }
  }
}
```

With the error message being:

```
  Description: [CT.LAMBDA.PR.2]: Require AWS Lambda function policies to prohibit public access
  How to fix: [FIX]: When setting 'Principal' to '*', provide one of 'SourceAccount', 'SourceArn', or 'PrincipalOrgID'. When setting 'Principal' to a service principal (for example, s3.amazonaws.com), provide one of 'SourceAccount' or 'SourceArn'.
```

Do you know why this may be happening, given that the rule is also enabled in the control tower, but deployments there are normally successful?

Thanks!
